src/engines/PDFEngine.js

```diff
@@ -1,33 +1,73 @@
 // @flow
 
-import * as htmlPdf from "html-pdf-chrome";
+import _ from "lodash";
+import chromium from "chrome-aws-lambda";
 import fs from "fs";
 import handlebars from "handlebars";
-import path from "path";
 import uuid from "uuid";
 
 class PDFEngine implements DocumentGeneratorEngine {
-  process = async (data: mixed, pathToTemplate: string) => {
-    const html = this.interpolateTemplate(data, pathToTemplate);
-    const [pdf, fileName] = await this.createFile(html);
+  process = async (data: mixed, templates: Array<string>) => {
+    const browser = await this.initPuppeteer();
 
-    return [pdf, fileName];
+    const files = await Promise.all(
+      _.map(templates, async pathToTemplate => {
+        const html = this.interpolateTemplate(data, pathToTemplate);
+
+        return this.createFile(html, browser);
+      })
+    );
+
+    // clean up
+    await browser.close();
+
+    return files;
   };
 
   interpolateTemplate = (data: mixed, pathToTemplate: string) => {
-    const templateFile = fs.readFileSync(path.join(__dirname, pathToTemplate));
+    const templateFile = fs.readFileSync(pathToTemplate);
     const template = handlebars.compile(templateFile.toString());
     const html = template(data);
 
     return html;
   };
 
-  createFile = async (html: string) => {
-    const fileName = `${uuid.v4()}.pdf`;
-    const pdf = await htmlPdf.create(html, { port: 9222 });
+  createFile = async (html: string, browser: any) => {
+    // start puppeteer or use the one provided
+    const page = await browser.newPage();
+
+    // render HTML
+    await page.setContent(html);
+
+    // TODO: pass default css to make it look like legal documents
```

Comment:
Why is this a TODO? because I think in our templates we already define a CSS under the `style` tag.

Reply:
oh really? that's exactly what I was looking to do, I'll just remove the comment then